pkg/utils/helper.go

```diff
@@ -132,28 +135,57 @@ func GetHostName() (string, error) {
 	return hostName.HostName, nil
 }
 
-// GetAPIServerFQDN gets the API Server FQDN from the kubeconfig file
-func GetAPIServerFQDN() (string, error) {
-	output, err := RunCommandOnHost("cat", "/var/lib/kubelet/kubeconfig")
-
-	if err != nil {
-		return "", fmt.Errorf("Can't open kubeconfig file: %+v", err)
-	}
-
+//ParseAPIServerFQDNFromKubeConfig parses a kubeConfig and returns the APIServerFQDN
+func ParseAPIServerFQDNFromKubeConfig(output string) (string, error) {
 	lines := strings.Split(output, "\n")
 	for _, line := range lines {
 		index := strings.Index(line, "server: ")
 		if index >= 0 {
 			fqdn := line[index+len("server: "):]
-			fqdn = strings.Replace(fqdn, "https://", "", -1)
-			fqdn = strings.Replace(fqdn, ":443", "", -1)
-			return fqdn, nil
+			fqdnurl, err := url.Parse(fqdn)
+			if err != nil {
+				return "", fmt.Errorf("parse url from fqdn: %s", fmt.Sprint(err)+": "+fqdn)
+			}
+
+			host, _, err := net.SplitHostPort(fqdnurl.Host)
+			if err != nil {
+				return "", fmt.Errorf("split host port from fqdnurl: %s: %w", fqdnurl, err)
+			}
+
+			return host, nil
 		}
 	}
-
 	return "", errors.New("Could not find server definitions in kubeconfig")
 }
 
+//ReadKubeletConfig reads the kubeletConfig from the node
+func ReadKubeletConfig() (string, error) {
```

Comment:
Hi @Tatsinnit, I have not tested periscope against GKE or EKS, so am not aware of the kubeconfig file locations for these distros.

Reply:
Agree - pulling out checkIfKubeConfigFileExist as a separate func is better. Can't use os.stat because this has to run on the Host - can be refactored to use a volume mount later.

Reply:
refactor to pull out func and use const's done @Tatsinnit 

Actually /etc/kubernetes/kubelet.conf might be the standard location as I think all the other distros (ARO, AzureStack etc) have it located here and AKS is the odd-one-out.